client/trino-jdbc/src/test/java/io/trino/jdbc/TestTrinoDriverAuth.java

```diff
@@ -256,6 +256,100 @@ public void testSuccessFullSslVerification()
         }
     }
 
+    @Test
+    public void testSuccessFullSslVerificationAlternateHostname()
+            throws Exception
+    {
+        String accessToken = newJwtBuilder()
+                .setSubject("test")
+                .setHeaderParam(KEY_ID, "33")
+                .signWith(privateKey33)
+                .compact();
+
+        String url = format("jdbc:trino://127.0.0.1:%s", server.getHttpsAddress().getPort());
+        Properties properties = new Properties();
+        properties.setProperty("SSL", "true");
+        properties.setProperty("SSLVerification", "FULL");
+        properties.setProperty("SSLTrustStorePath", new File(getResource("localhost.truststore").toURI()).getPath());
+        properties.setProperty("SSLTrustStorePassword", "changeit");
+        properties.setProperty("accessToken", accessToken);
+        properties.setProperty("hostnameInCertificate", "localhost");
+
+        try (Statement statement = DriverManager.getConnection(url, properties).createStatement()) {
+            assertTrue(statement.execute("SELECT 123"));
+            ResultSet rs = statement.getResultSet();
+            assertTrue(rs.next());
+            assertEquals(rs.getLong(1), 123);
+            assertFalse(rs.next());
+        }
+    }
+
+    @Test
+    public void testFailedFullSslVerificationAlternateHostnameNotProvided()
+            throws Exception
+    {
+        String accessToken = newJwtBuilder()
+                .setSubject("test")
+                .setHeaderParam(KEY_ID, "33")
+                .signWith(privateKey33)
+                .compact();
+
+        String url = format("jdbc:trino://127.0.0.1:%s", server.getHttpsAddress().getPort());
+        Properties properties = new Properties();
+        properties.setProperty("SSL", "true");
+        properties.setProperty("SSLVerification", "FULL");
+        properties.setProperty("SSLTrustStorePath", new File(getResource("localhost.truststore").toURI()).getPath());
+        properties.setProperty("SSLTrustStorePassword", "changeit");
+        properties.setProperty("accessToken", accessToken);
+
+        try (Statement statement = DriverManager.getConnection(url, properties).createStatement()) {
```

Comment:
Connection and Statement needs to be closed. By merging `try` statements I meant:
```
try (Connection connection = DriverManager.getConnection(url, properties);
      Statement statement = connection.createConnection()) {
```

I have noticed that this how other tests are implemented in this class. So can you please that other try blocks are also improved in preparatory commit before the main change of your PR?